src/Kiota.Builder/Writers/CSharpWriter.cs

```diff
@@ -95,28 +95,47 @@ public override void WriteMethod(CodeMethod code)
             var staticModifier = code.IsStatic ? "static " : string.Empty;
             var returnType = GetTypeString(code.ReturnType);
             // Task type should be moved into the refiner
-            WriteLine($"{GetAccessModifier(code.Access)} {staticModifier}async Task<{returnType}> {code.Name}({string.Join(", ", code.Parameters.Select(p=> GetParameterSignature(p)).ToList())}) {{");
+            WriteLine($"{GetAccessModifier(code.Access)} {staticModifier}{(code.IsAsync ? "async Task<": string.Empty)}{returnType}{( code.IsAsync ? ">": string.Empty)} {code.Name}({string.Join(", ", code.Parameters.Select(p=> GetParameterSignature(p)).ToList())}) {{");
             IncreaseIndent();
+            var requestBodyParam = code.Parameters.FirstOrDefault(x => x.ParameterKind == CodeParameterKind.RequestBody);
+            var queryStringParam = code.Parameters.FirstOrDefault(x => x.ParameterKind == CodeParameterKind.QueryParameter);
+            var headersParam = code.Parameters.FirstOrDefault(x => x.ParameterKind == CodeParameterKind.Headers);
             switch(code.MethodKind) {
```

Comment:
At some point we should refactor this to be separate methods.

Reply:
too javascripty for your taste? 😂 writers are getting big, maybe we should consider break them into a visitor pattern?

Reply:
resolving as we queued #83